rust-tide-backend/src/handler/grocery.rs

```diff
@@ -0,0 +1,33 @@
+use super::util::HandlerResponse;
+use crate::{data::grocery::*, error::Error::*, Request, Result};
+use chrono::Utc;
+use log::info;
+use tide::Response;
+
+pub async fn list(_req: Request) -> HandlerResponse {
+    HandlerResponse::new(do_list().await)
+}
+
+pub async fn create_list(req: Request) -> HandlerResponse {
+    HandlerResponse::new(do_create_list(req).await)
+}
+
+pub async fn do_list() -> Result<Response> {
+    let grocery_list = GroceryList {
+        name: String::from("example list"),
+        creation_date: Utc::now(),
+        items: vec![GroceryItem {
+            name: String::from("tea"),
+            amount: Some(1),
+        }],
+    };
+    let res = Response::new(200).body_json(&grocery_list)?;
```

Comment:
```suggestion
    Response::new(200).body_json(&grocery_list)
```
could be returned immediately or am I missing something?

Reply:
it swallows the json serialization error, so you would at least need to map the error

Reply:
Ah is `?` taking care of the error conversion automatically!?

Reply:
I think it tries to .into() it. you can for example do something like this:

```
    Response::new(200).body_json(&grocery_list).map_err(|e| e.into())
```